### PR TITLE
JHipster Deep Dive Part One

### PR DESCRIPTION
**Pull request by BitBruce**
One post turned into a series of three to four posts. Was initially going for a total and complete newbie-friendly guide to JHipster and Spring Boot projects in general, but was too much. Decided on complex entity examples and deployment examples, but first we must generate an application before we play with one, so part one will be "boring" but necessary.

I'm redoing a flowchart for the generator options section and adding some detail to each of the testing framework options. Will update first thing tomorrow/Wednesday.

I appreciate any remarks, feedback, suggestions. I haven't written in a while, and am starting to get the hang of it again. I have many blog post ideas now.

### CODE REVIEW

**Comment by BitBruce** [comment]
Re-opened, ready for review.

File: posts/jhipster-deepdive-part1.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+> [Learn how to configure your IDE for JHipster here!](https://www.jhipster.tech/configuring-ide/)
+
+In the next post we will dive deep into JHipster's default tools and generating several entities with relationships of various complexities.
+
+---
+
+*At the time of this writing, the current version of JHipster is v4.14.4 and the next major version, v5.0, is in beta. Version 5 will bring official support for another frontend framework [React](https://reactjs.org/) among many improvements. A follow-up post will be written for JHipster 5's release. This series is written using a MacBook Pro running High Sierra. Some steps will be different for other operating systems.*
```
Comment (raphaelbrugier) [inline posts/jhipster-deepdive-part1.md]: JHipster 5 is coming soon, I think the article should target it directly.

File: posts/jhipster-deepdive-part1.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Next we will look at the various options in JHipster's application generator. Note that unless development is done inside a Docker or Vagrant container (not highly recommended), you will need to have installed [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html), [Git](https://git-scm.com/downloads), [Node.js](https://nodejs.org/en/download/), [Yarn](https://yarnpkg.com/en/) (or [NPM](https://www.npmjs.com/get-npm)) and [Yeoman](http://yeoman.io/) installed, in addition to the database you will choose. Optionally, install [Docker](https://docs.docker.com/install/) for running the database inside of a container instead of on your operating system (I personally prefer managing my databases inside containers).
```
Comment (dsharpe) [inline posts/jhipster-deepdive-part1.md]: Remove "installed" after Yeoman

File: posts/jhipster-deepdive-part1.md (hunk 1 above)
Comment (dsharpe) [reply posts/jhipster-deepdive-part1.md]: I hate to wait. That just gives us an excuse to write another blog. 😄

File: posts/jhipster-deepdive-part1.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Getting Started
+
+First, we need to generate an application before we can get our hands dirty creating entities and exploring what more JHipster has to offer. Under the hood, JHipster uses a custom [Yeoman](http://yeoman.io/) generator to scaffold a Spring Boot application with choices of frontend framework, database, build tool, authentication type, testing framework and more. There are also several community-created modules that can be included during initial setup or imported later.
+
+> ["Why should organizations use JHipster?" by Mallik Sambaraju ](https://blog.ippon.tech/use-of-jhipster-in-organizations/)
```
Comment (aseigneurin) [inline posts/jhipster-deepdive-part1.md]: Remove space before `]`

File: posts/jhipster-deepdive-part1.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# JHipster Deep Dive, Part One
```
Comment (aseigneurin) [inline posts/jhipster-deepdive-part1.md]: Can you add a metadata block? Look at other posts, e.g. https://github.com/ippontech/blog-usa/edit/master/posts/10-tips-and-tricks-for-cassandra.md

The title will be included in this block, so you will need to remove one `#` from your titles (e.g. `Getting started` will be `#`).

**Comment by BitBruce** [comment]
I saw other posts' headers were all two hashes, should I leave Getting Started with ## or change to # ?

**Comment by dsharpe** [comment]
@aseigneurin is there a certain pattern we are adhering to for the headers?

**Pull request merged**